hazelcast/src/main/java/com/hazelcast/config/QueueConfig.java

```diff
@@ -323,6 +327,57 @@ public QueueConfig setMergePolicyConfig(MergePolicyConfig mergePolicyConfig) {
         return this;
     }
 
+    /**
```

Comment:
This doesn't seem very intuitive. Because a comparator is set, it becomes a priority queue.

Reply:
It's how we avoid yet another config class and another interface and another method on `HazelcastInstance`. I don't mind, as long as the documentation is clear. Any other suggestions for exposing a priority queue?

Reply:
I do mind because it isn't intuitive. I think having an explicit property on it like 'boolean priorityQueue' makes more sense en have a default comparator implementation be in place that makes use of the Comparable interface. So the only thin someone should need to do is 'queueConfig.setPriorityQueue(true)....'

Reply:
@pveentjer
I agree that API should be intuitive for usage. What's your take on below suggestions?

Reply:
Following suggestions done above, I went for this:
```<queue name="myQueue" queue-type="PRIORITY_QUEUE" comparator-class-name="com.mycompany.MyQueueItemSpecificComparator" duplicate-allowed=false>
</queue>```